ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt

```diff
@@ -77,6 +77,17 @@ interface UtxoLedgerService {
     @Suspendable
     fun filterSignedTransaction(signedTransaction: UtxoSignedTransaction): UtxoFilteredTransactionBuilder
 
+    /**
+     * Finds unconsumed states of transaction in the vault, resolves it to a ledger transaction and returns it
+     *
+     * @param id The id of the transaction to find
+     * @param stateClass State type
+     *
+     * @return The unconsumed states (or empty list if not found).
+     */
+    @Suspendable
+    fun <T: ContractState> findUnconsumedStatesByType(id: SecureHash, stateClass: Class<out T>): List<StateAndRef<T>>
```

Comment:
Why are we looking by transaction id? I thought idea was to find any consumable states of a certain type, whithout knowing their transaction id?

Reply:
fixed